### ISSUE TITLE
Unspecific error messages when local Swift version not supported

### ISSUE THREAD

**Issue by simonseyer** [yonaskolb/Mint#106]
Example: Swiftgen defines Swift 4.2 as minimum version in their [`Package.swift`](https://github.com/SwiftGen/SwiftGen/blob/6.0.0/Package.swift#L45) file. On my local machine Swift 4.1.2 is active right now.

When I try to install `swiftgen/swiftgen@6.0.0` I get the error message: `Package.swift not found`. This error message is in general not too helpful but even wrong in this case, because the file is there but the version is incompatible.

The underlying reason is the output of the following line: [SwiftPackage.swift#L12](https://github.com/yonaskolb/Mint/blob/master/Sources/MintKit/SwiftPackage.swift#L12)

```
error: package at '/.../mint/github.com_swiftgen_swiftgen' 
requires a minimum Swift tools version of 4.2.0 (currently 4.1.0)
```

Suggestion: parse this error message and forward it to the users of mint.

**Comment by yonaskolb** [comment]
Thanks for opening @simonseyer.
From triaging this issue https://github.com/krzysztofzablocki/Sourcery/issues/689, it seems things can also fail if the minimum tools version is correct but other dependencies in the package require a different Swift version. Unfortunately that error doesn't actually say that, so we can't parse it. It just looks something like this:
```
error: dependency graph is unresolvable; found these conflicting requirements:

Dependencies:
    https://github.com/tadija/AEXML.git @ 4.3.3
'Sourcery' /Users/Yonas/Downloads/Sourcery-master: error: product dependency 'PathKit' not found
'Sourcery' /Users/Yonas/Downloads/Sourcery-master: error: product dependency 'Commander' not found
'Sourcery' /Users/Yonas/Downloads/Sourcery-master: error: product dependency 'PathKit' not found
'Sourcery' /Users/Yonas/Downloads/Sourcery-master: error: product dependency 'SourceKittenFramework' not found
'Sourcery' /Users/Yonas/Downloads/Sourcery-master: error: product dependency 'StencilSwiftKit' not found
'Sourcery' /Users/Yonas/Downloads/Sourcery-master: error: product dependency 'xcproj' not found
'Sourcery' /Users/Yonas/Downloads/Sourcery-master: error: product dependency 'SwiftTryCatch' not found
```

We might just have to have a generic error message saying `This package cannot be compiled with Swift \(swiftVersion)`

**Comment by yonaskolb** [comment]
Fixed in 0.11.3
